src/renderer/components/commands-action-button.tsx

```diff
@@ -259,21 +259,15 @@ export class GistActionButton extends React.Component<
     const { gistId } = this.props.appState;
     const { actionType } = this.state;
 
-    if (gistId) {
-      switch (actionType) {
-        case GistActionType.publish:
-          await this.handlePublish();
-          break;
-        case GistActionType.update:
-          await this.handleUpdate();
-          break;
-        case GistActionType.delete:
-          await this.handleDelete();
-          break;
-      }
-    } else {
-      await this.handlePublish();
+    if (gistId && actionType === GistActionType.update) {
+      return this.handleUpdate();
+    }
+
+    if (gistId && actionType == GistActionType.delete) {
```

Comment:
Is the `==` intentional here?

Reply:
Nope. Fixed 9a1a55bc95fe4085fb28fc1df2220ae6cbd5de64